cunumeric/array.py

```diff
@@ -2119,6 +2125,30 @@ def copy(self, order="C") -> ndarray:
         # We don't care about dimension order in cuNumeric
         return self.__copy__()
 
+    @add_boilerplate()
+    def cumsum(self, axis=None, dtype=None, out=None) -> ndarray:
+        return self._perform_scan(
+            ScanCode.SUM, self, axis=axis, dtype=dtype, out=out, nan0=False
```

Comment:
a nitpick: I'd change `nan0` to `nan_to_identity` instead, as you don't change NaNs to 0 in cumprod.

Reply:
90427886a52de634b883d2a0590ed15b65368064